app/models/article.rb

```diff
@@ -80,6 +80,8 @@ class Article < ApplicationRecord
 
   scope :cached_tagged_with, ->(tag) { where("cached_tag_list ~* ?", "^#{tag},| #{tag},|, #{tag}$|^#{tag}$") }
 
+  scope :cached_tagged_by_approval_with, ->(tag) { where("cached_tag_list ~* ? AND approved = true", "^#{tag},| #{tag},|, #{tag}$|^#{tag}$") }
```

Comment:
If I'm not mistaken this is the same as `cached_tagged_with` plus the `approved = true` condition:

```ruby
[1] pry(main)> Article.cached_tagged_with("discuss").where(approved: true).to_sql
=> "SELECT \"articles\".* FROM \"articles\" WHERE (cached_tag_list ~* '^discuss,| discuss,|, discuss$|^discuss$') AND \"articles\".\"approved\" = TRUE"
[2] pry(main)> Article.cached_tagged_by_approval_with("discuss").to_sql
=> "SELECT \"articles\".* FROM \"articles\" WHERE (cached_tag_list ~* '^discuss,| discuss,|, discuss$|^discuss$' AND approved = true)"
```

my proposal is to type less :D 

```ruby
  scope :cached_tagged_by_approval_with, ->(tag) { cached_tagged_with(tag).where(approved: true) }
```

which results in:

```ruby
[1] pry(main)> Article.cached_tagged_by_approval_with("discuss").to_sql
=> "SELECT \"articles\".* FROM \"articles\" WHERE (cached_tag_list ~* '^discuss,| discuss,|, discuss$|^discuss$') AND \"articles\".\"approved\" = TRUE"
```